src/soft-tabs-transform.coffee

```diff
@@ -0,0 +1,38 @@
+require('es6-shim')
+
+Point = require './point'
+Region = require './region'
+
+module.exports =
+class SoftTabsTransform
+  constructor: (@tabLength) ->
+    @softTab = " ".repeat(@tabLength)
+
+  initialize: (@source) ->
+
+  getRegions: ({sourceStartPosition}) ->
+    sourceEndPosition = @source.getEndPosition()
+    sourcePosition = sourceStartPosition
+    regions = []
```

Comment:
I'm sort of having regrets now about suggesting we return more than one region at a time here. Sorry for the trouble, but I take it back. Instead what we should do is define a `startsWith` method on layers that takes a string and an optional position. In this case we would simply ask if the underlying layer starts with a soft tab at our current position and if it does, consume it. Otherwise we'd return a region extending to the end of the line as you do. This ensures that we don't scan ahead any more than needed in the input, which will keep the streaming window as small as possible as we're constructing regions at each layer.

It's actually important that `getRegions` return the minimal number of regions because we'll end up calling it when regions get invalidated. I'm realizing that when `getRegions` returns multiple regions, we should probably link their invalidation together on a change event. So if a transform returns regions A, B, and C and at some point B there's a change in the source that collides with B, then C also gets invalidated. This is how we can propagate changes through soft-wraps. But in the case of soft-tabs, we don't really need such a linkage. Sorry for the thrash on this, but I'm still getting my head wrapped around the problem.